hkmc2/shared/src/test/mlscript/lifter/StackSafetyLift.mls

```diff
@@ -97,93 +76,59 @@ fun sum(n) =
     n + sum(n - 1)
 sum(10000)
 //│ JS (unsanitized):
-//│ let sum1;
-//│ let Cont$func$sum$1, res1, $_stack$_safe$_body$_1, doUnwind$1, Cont$func$sum$$;
-//│ Cont$func$sum$$ = function Cont$func$sum$$(isMut, n, tmp, pc) {
-//│   let tmp1, tmp2;
-//│   if (isMut === true) {
-//│     tmp1 = new Cont$func$sum$1(pc);
-//│   } else {
-//│     tmp1 = globalThis.Object.freeze(new Cont$func$sum$1(pc));
-//│   }
-//│   tmp2 = tmp1(n, tmp);
-//│   return tmp2
-//│ };
-//│ (class Cont$func$sum$ extends runtime.FunctionContFrame.class {
-//│   static {
-//│     Cont$func$sum$1 = this
-//│   }
-//│   constructor(pc) {
-//│     return (n, tmp) => {
-//│       let tmp1;
-//│       tmp1 = super(null);
-//│       this.n = n;
-//│       this.tmp = tmp;
-//│       this.pc = pc;
-//│       return this;
-//│     }
-//│   }
-//│   #n;
-//│   #tmp;
-//│   get n() { return this.#n; }
-//│   set n(value) { this.#n = value; }
-//│   get tmp() { return this.#tmp; }
-//│   set tmp(value) { this.#tmp = value; }
-//│   resume(value$) {
-//│     let curDepth;
-//│     curDepth = runtime.stackDepth;
-//│     if (this.pc === 1) {
-//│       this.tmp = value$;
-//│     }
-//│     contLoop: while (true) {
-//│       runtime.stackDepth = curDepth;
-//│       switch (this.pc) {
-//│         case 0:
-//│           return sum1(this.n);
-//│           break;
-//│         case 1:
-//│           return this.n + this.tmp;
-//│           break;
-//│       }
-//│       break;
-//│     }
-//│   }
-//│   toString() { return runtime.render(this); }
-//│   static [definitionMetadata] = ["class", "Cont$func$sum$"]; 
-//│ });
-//│ doUnwind$1 = function doUnwind$(n, tmp, res2, pc) {
-//│   res2.contTrace.last.next = Cont$func$sum$$(true, n, tmp, pc);
-//│   res2.contTrace.last = res2.contTrace.last.next;
-//│   return res2
-//│ };
+//│ let sum1, tmp1;
+//│ let $_stack$_safe$_body$_1;
 //│ sum1 = function sum(n) {
-//│   let scrut, tmp, tmp1;
-//│   let curDepth, stackDelayRes;
+//│   let scrut, tmp2, tmp3, pc;
+//│   let tmp4, curDepth, stackDelayRes;
 //│   curDepth = runtime.stackDepth;
 //│   runtime.stackDepth = runtime.stackDepth + 1;
 //│   stackDelayRes = runtime.checkDepth();
 //│   if (stackDelayRes instanceof runtime.EffectSig.class) {
-//│     return doUnwind$1(n, tmp1, stackDelayRes, 0)
+//│     return runtime.unwind(stackDelayRes, 1, sum1, null, "StackSafetyLift.mls:73:1", 0, null, 2, n, tmp2)
 //│   }
-//│   scrut = n === 0;
-//│   if (scrut === true) {
-//│     return 0
+//│   if (runtime.isResuming === true) {
```

Comment:
Couldn't we save a bit of time (fewer accesses) by using something like `let rpc = runtime.resumePc; if (rpc < 0) { ... }` instead of `if (runtime.isResuming) === true { ... }`?